optimum/intel/openvino/trainer.py

```diff
@@ -536,18 +667,77 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
             output_path = os.path.join(output_dir, OV_XML_FILE_NAME)
             self.compression_controller.prepare_for_export()
             model_type = self.model.config.model_type.replace("_", "-")
-            onnx_config_cls = FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][self.feature]
-            onnx_config = onnx_config_cls(self.model.config)
-            use_external_data_format = (
-                onnx_config.use_external_data_format(self.model.num_parameters()) or self.ov_config.save_onnx_model
-            )
-            f = io.BytesIO() if not use_external_data_format else os.path.join(output_dir, "model.onnx")
-            self._onnx_export(self.model, onnx_config, self.ov_config, f)
+            if self.onnx_config is None:
+                onnx_config_cls = FeaturesManager._SUPPORTED_MODEL_TYPE[model_type][self.feature]
+                onnx_config = onnx_config_cls(self.model.config)
+            else:
+                onnx_config = self.onnx_config
+
+            if self._is_pruning_controller_exists():
+                # Note:
+                # OpenVINO provides automated structured pruning of sparse structure in IR.
+                # However it requires static axes, current export utilizes nncf exporter
+                # which generates static-shaped IR.
 
-            # Load and save the compressed model
-            model = core.read_model(f) if use_external_data_format else core.read_model(f.getvalue(), b"")
-            compress_quantize_weights_transformation(model)
-            openvino.runtime.serialize(model, output_path, output_path.replace(".xml", ".bin"))
+                f = os.path.join(output_dir, "model.onnx")
```

Comment:
[`ONNX_WEIGHTS_NAME`](https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/utils.py#L21) could be used instead of "model.onnx" (should have been done before)

Reply:
fixed in https://github.com/huggingface/optimum-intel/pull/150/commits/76991dcb79da98ea520a644e42f840a05c790ae4